WelsonJS.Toolkit/WelsonJS.Launcher/TraceLogger.cs

```diff
@@ -30,8 +32,27 @@ static TraceLogger()
             Trace.AutoFlush = true;
         }
 
-        public void Info(string message) => Trace.TraceInformation(message);
-        public void Warn(string message) => Trace.TraceWarning(message);
-        public void Error(string message) => Trace.TraceError(message);
+        public void Info(params object[] args) => Trace.TraceInformation(Format(args));
+        public void Warn(params object[] args) => Trace.TraceWarning(Format(args));
+        public void Error(params object[] args) => Trace.TraceError(Format(args));
+
+        private static string Format(object[] args)
+        {
+            if (args == null || args.Length == 0) return string.Empty;
+
+            if (args.Length == 1)
+                return args[0]?.ToString() ?? string.Empty;
+
+            string format = args[0]?.ToString() ?? string.Empty;
+            try
+            {
+                return string.Format(format, args.Skip(1).ToArray());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Format string is constructed from untrusted input received from network or file streams. An attacker could inject format string patterns leading to information disclosure or application crashes. Use string interpolation or explicit format parameters with fixed format strings instead of using external data directly in string formatting operations. For more information: https://owasp.org/www-community/attacks/Format_string_attack